vuln_checker/main.py

```diff
@@ -172,6 +174,83 @@ def combine_vulnerabilities(vulns: VulnMap) -> VulnMap:
     return traverse_graph_and_combine_vulnerabilities(graph)
 
 
+def dump_vulns(vulns: VulnMap):
+    with (VULNS_PATH / "combined.json").open("w") as f:
+        json.dump(vulns, f, default=set_to_list, indent=2)
+
+
+Advisory = TypedDict(
+    'Advisory', {
+        'source': str,
+        'id': str,
+        'severity': str | None,
+        'cvss_v3_vector': str | None,
+        'cvss_v3_score': float | None
+    }
+)
+
+GroupedAdvisories = dict[str, list[Advisory]]
+DifferentAdvisories = TypedDict(
+    'DifferentAdvisories', {'severity': list[GroupedAdvisories]}
+)
+
+KeyForCheck = Literal['severity', 'cvss_v3_vector', 'cvss_v3_score']
+
+
+def compare_advisory_by_field_name(advisories: Iterable[Advisory], field_name: KeyForCheck) -> GroupedAdvisories:
+    key = itemgetter(field_name)
+    advisories = [advisory for advisory in advisories if advisory[field_name]]
+
+    sorted_advisories = sorted(advisories, key=key)
+    grouped = itertools.groupby(sorted_advisories, key)
+    return {k: list(g) for (k, g) in grouped}
+
+
+def advisory_to_short_info(advisory: Advisory) -> str:
+    source = advisory["source"].split("/")
+    identifier = source[-1][:-5]
+    database = source[-2]
+    return f"{identifier} in {database}"
+
+
+keys_for_check: tuple[Literal['severity']] = (
+    "severity",
+)
+
+
+def find_different_advisories(vulns: VulnMap) -> DifferentAdvisories:
+    different_advisories = {k: [] for k in keys_for_check}
```

Comment:
why do we need it?

something like `different_advisories.get(k, [])` looks better, right?

Reply:
but i'm not sure about this case

Reply:
@afdesk I'm just initializing the dictionary so that I don't have to check if an element exists when adding a new one.

Reply:
if i understand correctly we should use `collections.defaultdict` in such cases, right?
```suggestion
    different_advisories = defaultdict(list)
```